### DISCUSSION TITLE
Annotations should be sorted by severity (error, warning, notice)

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#61736] [Actions] [answered]
### Select Topic Area

Bug

### Body

Consider this run: https://github.com/jsoref/annotation-limits/actions/runs/5655942044

<details><summary>10 errors, 10 warnings, and 10 notices</summary>

Annotations
10 errors, 10 warnings, and 10 notices
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:11)
error 1
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:14)
error 2
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:17)
error 3
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:20)
error 4
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:23)
error 5
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:26)
error 6
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:29)
error 7
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:32)
error 8
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:35)
error 9
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:38)
error 10
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:9)
notice 1
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:12)
notice 2
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:15)
notice 3
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:18)
notice 4
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:21)
notice 5
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:24)
notice 6
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:27)
notice 7
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:30)
notice 8
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:33)
notice 9
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:36)
notice 10
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:10)
warning 1
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:13)
warning 2
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:16)
warning 3
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:19)
warning 4
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:22)
warning 5
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:25)
warning 6
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:28)
warning 7
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:31)
warning 8
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:34)
warning 9
[build](https://github.com/jsoref/annotation-limits/actions/runs/5655942044/job/15322024796#step:2:37)
warning 10

</details>

Fwiw, there's a per step per severity annotation limit of 10 (this is documented somewhere), and there's a per job limit of 50 total annotations (also documented _somewhere_). 

I can't think of _any_ **good** reason to sort `notice` between `error` and `warning`.

I put together a more random list because I wanted to understand a bit more about ordering:

https://github.com/jsoref/annotation-limits/actions/runs/5656281688

<details><summary>16 errors, 17 warnings, and 17 notices</summary>

Annotations
16 errors, 17 warnings, and 17 notices

[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:11)
error Phrygian 31
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:14)
error unatmospheric 45
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:17)
error clinohedrite 8
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:20)
error opposer 28
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:23)
error Eva 14
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:26)
error greenhearted 17
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:29)
error tachygraph 42
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:32)
error ascophore 3
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:35)
error ovigerm 29
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:38)
error nonpathogenic 27
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:11)
error barrico 4
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:14)
error curlike 10
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:17)
error albarello 1
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:20)
error fixed 15
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:23)
error Dob 12
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:26)
error unpredacious 47
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:9)
notice Phrygian 31
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:12)
notice unatmospheric 45
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:15)
notice clinohedrite 8
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:18)
notice opposer 28
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:21)
notice Eva 14
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:24)
notice greenhearted 17
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:27)
notice tachygraph 42
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:30)
notice ascophore 3
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:33)
notice ovigerm 29
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:36)
notice nonpathogenic 27
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:9)
notice barrico 4
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:12)
notice curlike 10
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:15)
notice albarello 1
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:18)
notice fixed 15
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:21)
notice Dob 12
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:24)
notice unpredacious 47
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:27)
notice cerebromeningeal 7
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:10)
warning Phrygian 31
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:13)
warning unatmospheric 45
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:16)
warning clinohedrite 8
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:19)
warning opposer 28
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:22)
warning Eva 14
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:25)
warning greenhearted 17
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:28)
warning tachygraph 42
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:31)
warning ascophore 3
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:34)
warning ovigerm 29
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:3:37)
warning nonpathogenic 27
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:10)
warning barrico 4
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:13)
warning curlike 10
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:16)
warning albarello 1
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:19)
warning fixed 15
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:22)
warning Dob 12
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:25)
warning unpredacious 47
[left](https://github.com/jsoref/annotation-limits/actions/runs/5656281688/job/15322994990#step:5:28)
warning cerebromeningeal 7

</details>

It is not simply taking the first n items, nor is it sorting them by their outputs (beyond the fact that it's sorting items based on their severity, alphabetically!).

Anyway, bottom line:

Please do not sort severities alphabetically. Severities should be sorted by severity.

**Comment by nebuk89** [comment, marked as answer]
We snuck this ship in yesterday as part of adding a job level view of annotations :)

**Reply by jsoref** [reply]
Thanks, here's a new run (the first run attempt crashed, which is fun):
https://github.com/jsoref/annotation-limits/actions/runs/11142780688

> Update README.md #9
Annotations
16 errors, 17 warnings, and 17 notices
left
succeeded 8 minutes ago in 0s

<Details><summary>log 1</summary>

```
0s
0s
Run for a in "isochlor 21" "hyomental 19" "contravindicate 9" "rhabdoid 36" "unexistence 46" "pressor 33" "sate 37" "Triangulum 44" "pectinic 30" "Polymyodi 32" ; do
Notice: notice isochlor 21
Warning: warning isochlor 21
Error: error isochlor 21
Notice: notice hyomental 19
Warning: warning hyomental 19
Error: error hyomental 19
Notice: notice contravindicate 9
Warning: warning contravindicate 9
Error: error contravindicate 9
Notice: notice rhabdoid 36
Warning: warning rhabdoid 36
Error: error rhabdoid 36
Notice: notice unexistence 46
Warning: warning unexistence 46
Error: error unexistence 46
Notice: notice pressor 33
Warning: warning pressor 33
Error: error pressor 33
Notice: notice sate 37
Warning: warning sate 37
Error: error sate 37
Notice: notice Triangulum 44
Warning: warning Triangulum 44
Error: error Triangulum 44
Notice: notice pectinic 30
Warning: warning pectinic 30
Error: error pectinic 30
Notice: notice Polymyodi 32
Warning: warning Polymyodi 32
Error: error Polymyodi 32
0s
```

</Details>
<Details><summary>log 2</summary>

```
Run for a in "Phrygian 31" "unatmospheric 45" "clinohedrite 8" "opposer 28" "Eva 14" "greenhearted 17" "tachygraph 42" "ascophore 3" "ovigerm 29" "nonpathogenic 27" ; do
Notice: notice Phrygian 31
Warning: warning Phrygian 31
Error: error Phrygian 31
Notice: notice unatmospheric 45
Warning: warning unatmospheric 45
Error: error unatmospheric 45
Notice: notice clinohedrite 8
Warning: warning clinohedrite 8
Error: error clinohedrite 8
Notice: notice opposer 28
Warning: warning opposer 28
Error: error opposer 28
Notice: notice Eva 14
Warning: warning Eva 14
Error: error Eva 14
Notice: notice greenhearted 17
Warning: warning greenhearted 17
Error: error greenhearted 17
Notice: notice tachygraph 42
Warning: warning tachygraph 42
Error: error tachygraph 42
Notice: notice ascophore 3
Warning: warning ascophore 3
Error: error ascophore 3
Notice: notice ovigerm 29
Warning: warning ovigerm 29
Error: error ovigerm 29
Notice: notice nonpathogenic 27
Warning: warning nonpathogenic 27
Error: error nonpathogenic 27
0s
0s
0s
0s
```

</Details>
<Details><summary>log 3</summary>

```
Run for a in "lacertiform 22" "subpunctuation 41" "septenniality 38" "squalor 40" "inenarrable 20" "puffin 34" "dermutation 11" "Naga 26" "gaminesque 16" "rechurn 35" ; do
Notice: notice lacertiform 22
Warning: warning lacertiform 22
Error: error lacertiform 22
Notice: notice subpunctuation 41
Warning: warning subpunctuation 41
Error: error subpunctuation 41
Notice: notice septenniality 38
Warning: warning septenniality 38
Error: error septenniality 38
Notice: notice squalor 40
Warning: warning squalor 40
Error: error squalor 40
Notice: notice inenarrable 20
Warning: warning inenarrable 20
Error: error inenarrable 20
Notice: notice puffin 34
Warning: warning puffin 34
Error: error puffin 34
Notice: notice dermutation 11
Warning: warning dermutation 11
Error: error dermutation 11
Notice: notice Naga 26
Warning: warning Naga 26
Error: error Naga 26
Notice: notice gaminesque 16
Warning: warning gaminesque 16
Error: error gaminesque 16
Notice: notice rechurn 35
Warning: warning rechurn 35
Error: error rechurn 35
0s
```

</Details>
<Details><summary>log 4</summary>

```
Run for a in "barrico 4" "curlike 10" "albarello 1" "fixed 15" "Dob 12" "unpredacious 47" "cerebromeningeal 7" "bleatingly 5" "meatotomy 24" "upshoot 48" ; do
Notice: notice barrico 4
Warning: warning barrico 4
Error: error barrico 4
Notice: notice curlike 10
Warning: warning curlike 10
Error: error curlike 10
Notice: notice albarello 1
Warning: warning albarello 1
Error: error albarello 1
Notice: notice fixed 15
Warning: warning fixed 15
Error: error fixed 15
Notice: notice Dob 12
Warning: warning Dob 12
Error: error Dob 12
Notice: notice unpredacious 47
Warning: warning unpredacious 47
Error: error unpredacious 47
Notice: notice cerebromeningeal 7
Warning: warning cerebromeningeal 7
Error: error cerebromeningeal 7
Notice: notice bleatingly 5
Warning: warning bleatingly 5
Error: error bleatingly 5
Notice: notice meatotomy 24
Warning: warning meatotomy 24
Error: error meatotomy 24
Notice: notice upshoot 48
Warning: warning upshoot 48
Error: error upshoot 48
0s
0s
```

</Details>
<Details><summary>log 5</summary>

```
Run for a in "cabalic 6" "antennate 2" "thiamide 43" "misreceive 25" "loin 23" "waldgrave 49" "embrasure 13" "slopely 39" "yuleblock 50" "heptamerous 18" ; do
Notice: notice cabalic 6
Warning: warning cabalic 6
Error: error cabalic 6
Notice: notice antennate 2
Warning: warning antennate 2
Error: error antennate 2
Notice: notice thiamide 43
Warning: warning thiamide 43
Error: error thiamide 43
Notice: notice misreceive 25
Warning: warning misreceive 25
Error: error misreceive 25
Notice: notice loin 23
Warning: warning loin 23
Error: error loin 23
Notice: notice waldgrave 49
Warning: warning waldgrave 49
Error: error waldgrave 49
Notice: notice embrasure 13
Warning: warning embrasure 13
Error: error embrasure 13
Notice: notice slopely 39
Warning: warning slopely 39
Error: error slopely 39
Notice: notice yuleblock 50
Warning: warning yuleblock 50
Error: error yuleblock 50
Notice: notice heptamerous 18
Warning: warning heptamerous 18
Error: error heptamerous 18
0s
```

</Details>

![Screenshot_20241002-072854](https://github.com/user-attachments/assets/40222c62-ca6d-4450-aba2-a4b2fbae0256)
![Screenshot_20241002-072907](https://github.com/user-attachments/assets/551cc957-9d89-4f56-b1f2-aa3c1be52504)
![Screenshot_20241002-072919](https://github.com/user-attachments/assets/301bc790-b852-474a-9c47-1e9e5c140176)
![Screenshot_20241002-072934](https://github.com/user-attachments/assets/3655182f-75e0-4072-a56e-4ca0b4f319a1)
![Screenshot_20241002-072943](https://github.com/user-attachments/assets/024803d2-58ef-4802-bc02-952867bac14a)

I'll not sure I can explain which log entries were taken...  I think it's most of log 2 and 1 from log 4?

**Reply by nebuk89** [reply]
That's a lot of entries!! I will have to go ask folks what logic we use here..
